docs/dg/dev/architecture/api-platform.md

```diff
@@ -0,0 +1,426 @@
+
+### Merchant Portal API
+
+- **API Type:** `merchant-portal`
+- **Application:** MerchantPortal
+- **Base URL:** `http://mp.glue.eu.spryker.local/`
+- **Use cases:** Marketplace merchant interfaces
+- **Example:** `/products`
+
+## Multi-layer schema merging
+
+One of the key features is support for multi-layer schema definitions that automatically merge:
+
+**Core layer** (vendor/spryker):
+
+```yaml
+resource:
+  name: Customers
+  properties:
+    email:
+      type: string
+```
+
+**Feature layer** (src/SprykerFeature):
+
+```yaml
+resource:
+  name: Customers
+  properties:
+    loyaltyPoints:
+      type: integer
+```
+
+**Project layer** (src/Pyz):
+
+```yaml
+resource:
+  name: Customers
+  properties:
+    email:
+      required: true  # Override core
+    customField:
+      type: string    # Project-specific
+```
+
+**Result**: A single merged resource with all properties, project layer taking precedence.
+
+## Integration with Spryker architecture
+
+### Dependency Injection
+
+API Platform fully integrates with Symfony Dependency Injection:
+
+```php
+// config/Zed/ApplicationServices.php
+$services->load('Pyz\\Zed\\', '../../../src/Pyz/Zed/');
+```
+
+Providers and Processors are automatically discovered and can use constructor injection:
+
+```php
+class CustomerBackofficeProvider implements ProviderInterface
+{
+    public function __construct(
+        private CustomerFacadeInterface $customerFacade,
+        private CustomerRepositoryInterface $customerRepository,
+    ) {}
+}
+```
+
+### Facade integration
+
+Resources can leverage existing Spryker facades:
+
+```php
+class CustomerBackendProcessor implements ProcessorInterface
+{
+    public function __construct(
+        private CustomerFacadeInterface $customerFacade,
+    ) {}
+
+    public function process(mixed $data, Operation $operation, ...): mixed
+    {
+        $customerTransfer = $this->mapToTransfer($data);
+        $response = $this->customerFacade->createCustomer($customerTransfer);
+        return $this->mapToResource($response->getCustomerTransfer());
+    }
+}
+```
+
+## Resource generation
+
+### Console commands
+
+```bash
+# Generate all configured API types
+docker/sdk glue api:generate
+
+# Generate specific API type
+docker/sdk glue api:generate backend
+
+# Validate schemas only
+docker/sdk glue api:generate --validate-only
+
+# Force regeneration (bypass cache)
+docker/sdk glue api:generate --force
+
+# Dry run
+docker/sdk glue api:generate --dry-run
+```
+
+### Debug commands
+
+```bash
+# List all resources
+docker/sdk glue api:debug --list
+
+# Inspect specific resource
+docker/sdk glue api:debug customers --api-type=backend
+
+# Show merged schema
+docker/sdk glue api:debug customers --api-type=backend --show-merged
+
+# Show contributing files
+docker/sdk glue api:debug customers --api-type=backend --show-sources
+```
+
+## Features
+
+### Automatic OpenAPI documentation
+
+API Platform generates interactive OpenAPI documentation:
```

Comment:
which root?
How to disable this on production? How to password-protect this on production?

Reply:
also, will it include ALL resources of that API, or only the API Platform ones?

Reply:
Added note about configuring this. The docs page from the API Platform shows only what it can see and does not render schema files from the existing APIs.